packages/core/src/types.ts

```diff
@@ -678,6 +678,15 @@ export interface MachineConfig<
    * The machine's own version.
    */
   version?: string;
+  schema?: MachineSchema<TContext, TEvent>;
+}
+
+export interface MachineSchema<TContext, TEvent extends EventObject> {
+  context?: TContext;
```

Comment:
I was wondering how exactly this will interact with the top-level context property and according to my [simple experiments](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLKIBaggZQRiAWzgB4AVAYQHtwIAPMAPmQG8AoZZBGyBgLmSUe9MABoOyAM74icAPwD2nTt1oMFg6mrASAvm31sYAVxAIwwGlygQ4kPAWLlGACmmP5AuCACeASgEyVgkbMGMoECkZYmQ4SUEAbgM2I1NzS0iEGzsIdHxsci1eJhc4ATysEFxo0iFtRj9WZFDwyLhkQyzbSArsFyUuYX5g5UHTSCgBABYAJj1xTndZRQkVIbABLpyHWRIWZBgqKgFJMChQAHMO1z89A1u2LZ7MPoHVYpXR7nHoabnOXQLKIeT7Kd4iTbZew1PYHI4nM6XURjWiTZAgYyEABG0GuLluAPuKSeuReVX6EnBwwGax+aNm8wkS2IoLW2kh3WqHlhhyoGlO5xAF2R31RAgx2Nxuhud10fiAA) this will work perfectly (like expected). 

In all cases TS favours the `schema.context` type which I suspect is done because:
- an explicit~ type is provided there through `createSchema`, so this is preferred over inferring the type of the top-level context
- inference for inner call (like the presented `identity`) has to happen first and by then the type is "explicit" when TS starts inferring for the whole `createMachine` call which leads us to the same situation as in the previous point

Reply:
Yep, I was pleasantly surprised by this too. Good to know!